src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirExecutableGeneratorTests.cs

```diff
@@ -0,0 +1,116 @@
+            binDirectory.Create();
+            libraryDirectory = new DirectoryInfo($"{prefix}-library");
+            libraryDirectory.Create();
+            libraryFiles = new List<FileInfo>()
+            {
+                CreateFile("lib1", libraryDirectory, "lib1 contents"),
+                CreateFile("lib2", libraryDirectory, "lib2 contents"),
+            };
+            includeDirectory = new DirectoryInfo($"{prefix}-include");
+            includeDirectory.Create();
+            sourceDirectory = new DirectoryInfo($"{prefix}-source");
+            sourceDirectory.Create();
+            sourceFiles = new List<FileInfo>()
+            {
+                CreateFile("src1.cpp", sourceDirectory, "src1 contents"),
+                CreateFile("src2.bc", sourceDirectory, "src2 contents"),
+            };
+            linkLibraries = new List<string> { "lib1", "lib2" };
+        }
+
+        public void Dispose()
+        {
+            Util.DeleteDirectory(sourceDirectory);
+            Util.DeleteDirectory(includeDirectory);
+            Util.DeleteDirectory(libraryDirectory);
+            Util.DeleteDirectory(binDirectory);
+        }
+
+        [Fact]
+        public async Task TestGenerateExecutable()
+        {
+            var executableFile = new FileInfo(Path.Combine(binDirectory.FullName, "executableFile"));
+            await executableGenerator.GenerateExecutableAsync(executableFile, sourceDirectory, new[] { libraryDirectory }, new[] { includeDirectory }, linkLibraries);
+
+            // Verify invocation of clang.
+            clangClientMock.Verify(obj => obj.CreateExecutableAsync(
+                It.Is<string[]>(s => s.OrderBy(val => val).SequenceEqual(sourceFiles.OrderBy(val => val.FullName).Select(fileInfo => fileInfo.FullName))),
+                It.Is<string[]>(s => s.OrderBy(val => val).SequenceEqual(linkLibraries.OrderBy(val => val))),
+                new[] { libraryDirectory.FullName },
+                new[] { includeDirectory.FullName },
+                executableFile.FullName));
+
+            // Verify files were copied.
+            Assert.True(FilesWereCopied(libraryFiles.ToArray(), binDirectory));
+        }
+
+        private static FileInfo CreateFile(string fileName, DirectoryInfo directory, string contents)
+        {
+            var filePath = Path.Combine(directory.FullName, fileName);
+            var fileInfo = new FileInfo(filePath);
+            using var fileStream = fileInfo.OpenWrite();
+            using var streamWriter = new StreamWriter(fileStream);
+            streamWriter.Write(contents);
+            return fileInfo;
+        }
+
+        private static bool FilesWereCopied(FileInfo[] files, DirectoryInfo destinationDirectory)
+        {
+            var destinationDirectoryFiles = destinationDirectory.GetFiles();
+            foreach (var file in files)
+            {
+                var copiedFile = destinationDirectoryFiles.FirstOrDefault(fileInfo => fileInfo.Name == file.Name);
```

Comment:
This conceals quadratic algorithm. It's OK as number of iterations is still small and reading file content prevails. Consider making comment about it. ex:
// Runtime is (Number of files in destination directory)*(number of FileInfo passed in).
 Also mention that file content is also compared, not just file names.

Reply:
Added comment and moved file content comparison to a method in Utils.cs.